### PR TITLE
clap4 & irx-config

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

**Comment by therealprof** [comment]
Looks okay in principle although I'm not super-psyched to rely on a crate which had a total of 453 downloads until now, a good fraction of which caused by this CI...

File: src/util.rs [hunk 1]
```diff
@@ -22,23 +22,47 @@ pub const BITS_PER_BYTE: u32 = 8;
 /// that are not valid in Rust ident
 const BLACKLIST_CHARS: &[char] = &['(', ')', '[', ']', '/', ' ', '-'];
 
+#[cfg_attr(feature = "bin", derive(serde::Deserialize))]
```
Comment (Emilgardis) [inline src/util.rs:25]: can we cfg this on `serde` instead

Comment (burrbull) [reply src/util.rs:25]: I've tried to use `serde`, but it doesn't compile. I don't know why.

Comment (Emilgardis) [reply src/util.rs:25]: it's probably because it's used as `dep:serde`, instead only do `serde` in the `features.bin = [...]`

**Review by Emilgardis — DISMISSED** [review]
looks good to me

The concern with popularity/usage of the `irx-config` is valid, but I think we can go ahead with this.

@therealprof what do you think?

**Comment by burrbull** [comment]
cc @Emilgardis Updated to clap 4.0

**Comment by therealprof** [comment]
@Emilgardis Whoops, sorry, didn't notice you left a comment a while ago. Yes, fine by me.
@burrbull If you could resolve the conflict...

**Comment by luojia65** [comment]
LGTM, maybe I should close my PR then...

**Comment by burrbull** [comment]
rebased

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+- Update `clap` to 3.2, use `irx-config` instead of `clap_conf`
```
Comment (luojia65) [inline CHANGELOG.md:10]: Should this be "update clap to 4.0"? :)

**Review by therealprof — APPROVED** [review]
Let's get the party going then...

bors r+

**Pull request merged**